src/cuda/cuda-shader-object-layout.cpp

```
@@ -124,10 +125,20 @@ inline size_t computeEntryPointParamsSize(slang::EntryPointReflection* entryPoin
 
-RootShaderObjectLayoutImpl::RootShaderObjectLayoutImpl(Device* device, slang::ProgramLayout* programLayout)
+RootShaderObjectLayoutImpl::RootShaderObjectLayoutImpl(
+    Device* device,
+    ShaderProgram* shaderProgram,
+    slang::ProgramLayout* programLayout
+)
     : ShaderObjectLayoutImpl(device, programLayout->getSession(), programLayout->getGlobalParamsTypeLayout())
     , m_programLayout(programLayout)
 {
+    SLANG_RHI_ASSERT(shaderProgram);
+    if (SLANG_FAILED(_addSyntheticResources(shaderProgram)))
```

Cppcheck (2.20.0)

[style] The function 'calcAligned' is never used.

(unusedFunction)